docs/static/playground/playground.js

```diff
@@ -1,404 +1,266 @@
-
-            var js = xhr(samplePath + '/sample.js').then(function (response) { return response.responseText });
-            js.then(function (_) {
-                LOADED_SAMPLES.push({
-                    id: sampleId,
-                    js: _
-                });
-                return callback(null, findLoadedSample(sampleId));
-            }, function (err) {
-                callback(err, null);
-            });
-        }
-
-        sampleSwitcher.onchange = function () {
-            var sampleId = sampleSwitcher.options[sampleSwitcher.selectedIndex].value;
-            window.location.hash = sampleId;
-        };
-
-        var playgroundContainer = document.getElementById('playground');
-
-        layout();
-        window.onresize = layout;
-
-        playgroundContainer.appendChild(sampleSwitcher);
-        playgroundContainer.appendChild(typingContainer);
-        playgroundContainer.appendChild(runContainer);
-
-        data.js.model = monaco.editor.createModel('console.log("hi")', 'typescript');
-
-        editor = monaco.editor.create(editorContainer, {
-            model: data.js.model,
-            minimap: {
-                enabled: false
+            if (myToken !== currentToken) {
+                return;
             }
+            editor.setValue(sample.js);
+            editor.setScrollTop(0);
+            sendMessage("importproject", sample.js);
         });
-
-        var autoRun = /autorun/.test(window.location.href);
-        if (autoRun) {
-            data.js.model.onDidChangeContent(function (e) {
-                if (!e.isFlush) {
-                    runDebounce();
-                }
-            });
-        }
-
-        var currentToken = 0;
-        function parseHash(firstTime) {
-            var sampleId = window.location.hash.replace(/^#/, '');
-            if (!sampleId) {
-                sampleId = PLAY_SAMPLES[0].id;
-            }
-
-            if (firstTime) {
-                for (var i = 0; i < sampleSwitcher.options.length; i++) {
-                    var opt = sampleSwitcher.options[i];
-                    if (opt.value === sampleId) {
-                        sampleSwitcher.selectedIndex = i;
-                        break;
-                    }
-                }
+    }
+    s.addEventListener("change", function (ev) {
+        var sampleId = s.options[s.selectedIndex].value;
+        window.location.hash = sampleId;
+    });
+    window.addEventListener("hashchange", parseHash);
+    parseHash(true);
+}
+function loadIframe(selected) {
+    if (selected === selectedEndpoint)
+        return;
+    for (var _i = 0, targets_2 = targets; _i < targets_2.length; _i++) {
+        var target = targets_2[_i];
+        for (var _a = 0, _b = target.endpoints; _a < _b.length; _a++) {
+            var endpoint = _b[_a];
+            if (!selected || selected === target.name + "-" + endpoint.name) {
+                iframe.setAttribute("src", endpoint.url);
```

Comment:
add ``ws=mem`` to avoid polluting the script list of the user.

Reply:
updated! i'll make a separate pr after this one to use editors.json, it's not a lot of code but i want to add some targets to the json so i might tweak streamer to add a list of disabled targets

Reply:
you can add a flag in editors.json to specify that it is supported in  particular tool...
```
"microbit": {
    "tutorial-tool": true,
    ...
}
```
to avoid interference